### PR TITLE
feat: concentrated liquidity swap milestone 3

### PR DESCRIPTION
**Pull request by czarcas7ic**
## What is the purpose of the change

Adds cross-tick swap logic

### CODE REVIEW

**Comment by mattverse** [comment]
Im wondering why there's so much merge conflicts? 🤔

File: x/concentrated-liquidity/pool.go [hunk 1]
```diff
@@ -254,7 +275,27 @@ func (k Keeper) CalcInAmtGivenOut(ctx sdk.Context, tokenOut sdk.Coin, tokenInDen
...
 		swapState.amountSpecifiedRemaining = swapState.amountSpecifiedRemaining.Sub(amountIn)
 		swapState.amountCalculated = swapState.amountCalculated.Add(amountOut.Quo(sdk.OneDec().Sub(swapFee)))
-		swapState.tick = priceToTick(sqrtPrice.Power(2))
+
+		if swapState.sqrtPrice.Equal(sqrtPrice) {
+			liquidityDelta, err := k.crossTick(ctx, p.Id, nextTick)
+			if err != nil {
+				return sdk.Coin{}, sdk.Coin{}, err
+			}
+			if lte {
```
Comment (mattverse) [inline x/concentrated-liquidity/pool.go:284]: ditto here for lte

File: x/concentrated-liquidity/pool.go [hunk 2]
```diff
@@ -254,7 +275,27 @@ func (k Keeper) CalcInAmtGivenOut(ctx sdk.Context, tokenOut sdk.Coin, tokenInDen
...
+				liquidityDelta = liquidityDelta.Neg()
+			}
+			swapState.liquidity = swapState.liquidity.Add(liquidityDelta.ToDec())
+			if swapState.liquidity.LTE(sdk.ZeroDec()) || swapState.liquidity.IsNil() {
+				return sdk.Coin{}, sdk.Coin{}, fmt.Errorf("no liquidity available, cannot swap")
+			}
+			if lte {
```
Comment (mattverse) [inline x/concentrated-liquidity/pool.go:291]: ditto here for lte

**Review by mattverse — COMMENTED** [review]
Left some comments and questions about this PR! Please take a look 🌮

**Review by mattverse — APPROVED** [review]
LGTM Thanks for addressing the comments! 🌮

**Review by p0mvn — APPROVED** [review]

**Pull request merged**